src/main/java/servlets/DataEntries.java

```diff
@@ -30,9 +32,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         numEntries = -1;
       }
     }
-    // Logic that goes here that calls two methods from the intermediate Database
-    // depending on what exactly
-    String json = DatabaseQuery.getJson(sharedObjects.dataInstance);
-    response.getWriter().println(json);
+    // get JSON from Database.
+    try {
+      Database database = new GCSDatabase(sharedObjects.databaseName);
+      String json = DatabaseQuery.getJson(database);
+      response.getWriter().println(json);
+    } catch (Exception e) {
```

Comment:
i reduced the code inside the try to just initializing the database (i had it like this before, but got compile time errors, but these errors arent shoeing anymore.) since it is only 1 line now in the try, does this meet the specific exception error message that you want?

Reply:
This change did reduce the code inside `try` to only the statements that can throw, which is good. By "catch a more specific exception," I mean that you should rarely write `catch (Exception e)`, and instead write `catch (IOException e)` or whatever specific subclass of `Exception` can be thrown by the statement. `Exception` can otherwise catch a lot of different things that you might not really want to catch.